src/internal/kopia/base_finder_test.go

```diff
@@ -325,629 +378,305 @@ func (suite *BaseFinderUnitSuite) TestNoResult_ErrorListingSnapshots() {
+					build(),
 			},
 			expectedBaseReasons: map[int][]identity.Reasoner{
 				1: testUser1Mail,
 			},
-			backupData: []backupInfo{
-				newBackupModel(testBackup2, true, false, false, nil, nil),
-				newBackupModel(testBackup1, true, true, false, nil, nil),
-			},
 		},
 		{
 			name:  "Old Backup Details Pointer",
 			input: testUser1Mail,
-			manifestData: []manifestInfo{
-				newManifestInfo(
-					testID1,
-					testT1,
-					testCompleteMan,
-					testBackup1,
-					nil,
-					testMail,
-					testEvents,
-					testUser1,
-					testUser2,
-					testUser3),
+			data: []baseInfo{
+				newBaseInfoBuilder(1, testT1, testAllUsersAllCats...).
+					legacyBackupDetails().
+					build(),
 			},
 			expectedBaseReasons: map[int][]identity.Reasoner{
 				0: testUser1Mail,
 			},
-			backupData: []backupInfo{
-				newBackupModel(testBackup1, true, true, true, nil, nil),
-			},
 		},
 		{
 			name:  "All One Snapshot With Merge Base",
 			input: testAllUsersAllCats,
-			manifestData: []manifestInfo{
-				newManifestInfo(
-					testID1,
-					testT1,
-					testCompleteMan,
-					testBackup1,
-					nil,
-					testMail,
-					testEvents,
-					testUser1,
-					testUser2,
-					testUser3),
+			data: []baseInfo{
+				newBaseInfoBuilder(1, testT1, testAllUsersAllCats...).
+					build(),
 			},
 			expectedBaseReasons: map[int][]identity.Reasoner{
 				0: testAllUsersAllCats,
 			},
-			backupData: []backupInfo{
-				newBackupModel(testBackup1, true, true, false, nil, nil),
-			},
 		},
 		{
 			name:  "All One Snapshot with Assist Base",
 			input: testAllUsersAllCats,
-			manifestData: []manifestInfo{
-				newManifestInfo(
-					testID1,
-					testT1,
-					testCompleteMan,
-					testBackup1,
-					nil,
-					testMail,
-					testEvents,
-					testUser1,
-					testUser2,
-					testUser3),
+			data: []baseInfo{
+				newBaseInfoBuilder(1, testT1, testAllUsersAllCats...).
+					setBackupType(model.AssistBackup).
+					build(),
 			},
 			expectedAssistReasons: map[int][]identity.Reasoner{
 				0: testAllUsersAllCats,
 			},
-			backupData: []backupInfo{
-				newBackupModel(
-					testBackup1,
-					true,
-					true,
-					false,
-					map[string]string{model.BackupTypeTag: model.AssistBackup},
-					nil),
-			},
 		},
 		{
 			name:  "Multiple Bases Some Overlapping Reasons",
 			input: testAllUsersAllCats,
-			manifestData: []manifestInfo{
-				newManifestInfo(
-					testID1,
-					testT1,
-					testCompleteMan,
-					testBackup1,
-					nil,
-					testMail,
-					testEvents,
-					testUser1,
-					testUser2,
-					testUser3),
-				newManifestInfo(
-					testID2,
-					testT2,
-					testCompleteMan,
-					testBackup2,
-					nil,
-					testEvents,
-					testUser1,
-					testUser2,
-					testUser3),
+			data: []baseInfo{
+				newBaseInfoBuilder(1, testT1, testAllUsersAllCats...).
+					build(),
+				newBaseInfoBuilder(2, testT2, testAllUsersMail...).
+					build(),
 			},
 			expectedBaseReasons: map[int][]identity.Reasoner{
```

Comment:
I swapped the reasons attached to bases at indices 0 and 1 because there's already a variable for all users mail but not one for all users events